dired-subtree.el

```diff
@@ -486,61 +489,62 @@ Return a string suitable for insertion in `dired' buffer."
 (defun dired-subtree-insert ()
   "Insert subtree under this directory."
   (interactive)
-  (let* ((dir-name (dired-get-filename nil))
-         (listing (dired-subtree--readin dir-name))
-         beg end)
-    (read-only-mode -1)
-    (move-end-of-line 1)
-    ;; this is pretty ugly, I'm sure it can be done better
-    (save-excursion
-      (insert listing)
-      (setq end (+ (point) 2)))
-    (newline)
-    (setq beg (point))
-    (let ((inhibit-read-only t))
-      (remove-text-properties (1- beg) beg '(dired-filename)))
-    (let* ((ov (make-overlay beg end))
-           (parent (dired-subtree--get-ov (1- beg)))
-           (depth (or (and parent (1+ (overlay-get parent 'dired-subtree-depth)))
-                      1))
-           (face (intern (format "dired-subtree-depth-%d-face" depth))))
-      (when dired-subtree-use-backgrounds
-        (overlay-put ov 'face face))
-      ;; refactor this to some function
-      (overlay-put ov 'line-prefix
-                   (if (stringp dired-subtree-line-prefix)
-                       (if (not dired-subtree-use-backgrounds)
-                           (apply 'concat (-repeat depth dired-subtree-line-prefix))
-                         (cond
-                          ((eq nil dired-subtree-line-prefix-face)
-                           (apply 'concat
-                                  (-repeat depth dired-subtree-line-prefix)))
-                          ((eq 'subtree dired-subtree-line-prefix-face)
-                           (concat
-                            dired-subtree-line-prefix
-                            (propertize
+  (when (dired-line-is-directory-or-link-p)
```

Comment:
This function looks like it was heavily modified, but I really just added this line and re-indented.